app/controllers/admin/topics_controller.rb

```diff
@@ -54,19 +54,21 @@ def update
 
   def destroy
     @topic = Topic.find(params[:id])
-    @questions = Question.where(topic_id: params[:id])
-      @questions.each do |q|
-        q.update(topic_id: 1)
-      end
-    @topic.destroy
-    flash[:success] = 'Topic deleted'
-    redirect_to admin_topics_path
+    unless @topic.id == 1
```

Comment:
i don't really understand  what the purpose of this code.

I can only guess that requirements is:
If topic has questions it should not be deletable. 
If so then code needs to be changed.